File: src/main/java/net/hectus/neobb/turn/legacy_game/block/LTTnt.java
package net.hectus.neobb.turn.legacy_game.block;

import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.default_game.attributes.clazz.RedstoneClazz;
import net.hectus.neobb.turn.default_game.attributes.function.AttackFunction;
import net.hectus.neobb.turn.default_game.block.BlockTurn;
import net.hectus.neobb.util.Modifiers;
import org.bukkit.block.Block;

import java.util.List;

public class LTTnt extends BlockTurn implements AttackFunction, RedstoneClazz {
    public LTTnt(NeoPlayer player) { super(player); }
    public LTTnt(Block data, NeoPlayer player) { super(data, player); }

    @Override
    public int cost() {
        return 2;
    }

    @Override
    public void apply() {
        List<NeoPlayer> eliminations = player.game.players().stream().filter(p -> !p.hasModifier(Modifiers.P_DEFAULT_DEFENDED)).toList();
        if (eliminations.size() >= player.game.players().size()) {
            player.game.draw(false);
        } else {
            eliminations.forEach(player.game::eliminatePlayer);
        }
    }
}

File: src/main/kotlin/net/hectus/neobb/modes/turn/legacy_game/block/LTTnt.kt
package net.hectus.neobb.modes.turn.legacy_game.block

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.modes.turn.default_game.attribute.clazz.RedstoneClazz
import net.hectus.neobb.modes.turn.default_game.attribute.function.AttackFunction
import net.hectus.neobb.modes.turn.default_game.block.BlockTurn
import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.util.Modifiers
import org.bukkit.block.Block

class LTTnt(data: Block?, cord: Cord?, player: NeoPlayer?) : BlockTurn(data, cord, player), AttackFunction, RedstoneClazz {
    override val cost: Int = 2

    override fun apply() {
        val eliminations = player!!.game.players.filter { !it.hasModifier(Modifiers.Player.Default.DEFENDED) }
        if (eliminations.size >= player.game.players.size) {
            player.game.draw()
        } else {
            eliminations.forEach { player.game.eliminate(it) }
        }
    }
}
